src/signalhandler.cpp

```diff
@@ -37,7 +39,11 @@ SignalHandler::SignalHandler() {
 }
 
 void SignalHandler::saHandler(int signal) {
-  logger.log() << "Signal" << signal;
   Q_ASSERT(self);
-  emit self->quitRequested();
+  logger.log() << "Signal" << signal << "from thread"
```

Comment:
I can't find a good explanation whether QMetaObject::invokeMethod() is async-safe with regards to POSIX signals so I am concerned that there is still a race condition here, and I have also encountered deadlocks in the logger during signal handling. I would recommend an approach closer to what's seen in mozilla-mobile/mozilla-vpn-client/pull/1190/

Reply:
Right! I revert this part and land the rest.